core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java

```diff
@@ -316,7 +316,7 @@ public FlowIdentifier updateNodeCountStartInstances(StackDto stackDto, InstanceG
                 updateNodeCountValidator.validateScalabilityOfInstanceGroup(stackDto, instanceGroupAdjustmentJson);
                 updateNodeCountValidator.validateScalingAdjustment(instanceGroupAdjustmentJson, stackDto);
                 if (withClusterEvent) {
-                    updateNodeCountValidator.validateClusterStatusForStartHostGroup(stack);
+                    updateNodeCountValidator.validateClusterStatusForStartHostGroup(stackDto);
```

Comment:
Shouldn't this check replace the existing `validateStackStatusForStartHostGroupCheck` happening above? In case the cluster is in NODE_FAILURE or any unfavourable state, the start / stop instances operations will error out because of that validation, rendering the RM health check useless.

Reply:
As discussed, the stackVersion check can be added to the relevant classes to determine if the scaling operation can proceed, as opposed to completely replacing the existing check

Reply:
Done